bolt-design-system/bolt
# Component Status Board + Foundation for API-specific Tasks

![image](https://user-images.githubusercontent.com/1617209/53053099-78d85080-346e-11e9-8a0e-e268b98d2899.png)


## Jira
http://vjira2:8080/browse/BDS-1046 and http://vjira2:8080/browse/BDS-744

## Summary
Adds an automatically generated (always up to date) component status board to Pattern Lab's main overview page.

## Details

### Pattern Lab Updates
- Patches Pattern Lab PHP logic to add a brand new `--dataonly` CLI config option to export the global PL data available (ex. nav URLs) without having to do a full Pattern Lab build

### Build-related Updates
- Moves the existing `bolt-versions` build task to new api-tasks folder in `@bolt/build-tools`
- Adds a new api-specific section to the build tasks as a new home for handling things such as:
    - Generating the data / component checks for the component status board + compiling the Twig via `@bolt/twig-renderer`
    - Generating visual regression testing URLs
    - Programmatically being able to find all available versions of the Bolt packages published to NPM 
- Updates the existing pattern lab build task to conditionally allow generating just the PL data (quick compile)
- Consolidates + improve coordination of the build and watch tasks for the new API-specific work vs the build and watch tasks for Pattern Lab (via events that get emitted) 

### Misc
- Updates the `iframe-resizer` library import path (used in a few places) to prevent build errors (build broken otherwise)


## How to test
- Review changes to make sure we're all on the same page
- Verify Travis builds and deploys as expected and the main PL overview page correctly displays the new component explorer
- Confirm Pattern Lab's watch task will rebuild and compile as expected
- Confirm changes to the status board's API-specific task in the build tools will regenerate when you change the template / data + subsequently recompile Pattern Lab after completed

> NOTE: a handful of components displayed in the component status table (ex. UL, OL and Button Group) are going to be slightly mismatched with the actual url in Pattern Lab due to some naming differences + bug in Pattern Lab PHP's logic with folder names.

I'm getting a separate PR going that'll work around that by renaming the problematic PL folders to match up with the component name / name we publish to NPM with.
